Tyep declarations shipped in 14.12.0 are invalid

### What steps are needed to reproduce the bug?

Write a custom rule importing stylelint
Run the typechecker on that file (which will import the stylelint types).
This will report an error in the stylelint type

### What Stylelint configuration is needed to reproduce the bug?

n/a

### How did you run Stylelint?

n/a

### Which version of Stylelint are you using?

14.12.0

### What did you expect to happen?

No type checking error

### What actually happened?

```
node_modules/stylelint/types/stylelint/index.d.ts:356:29 - error TS2344: Type 'RuleMessage' does not satisfy the constraint '(...args: any) => any'.
  Type 'string' is not assignable to type '(...args: any) => any'.

356    messageArgs?: Parameters<RuleMessage> | undefined;
                                ~~~~~~~~~~~


Found 1 error in node_modules/stylelint/types/stylelint/index.d.ts:356
```

### Does the bug relate to non-standard syntax?

_No response_

### Proposal to fix the bug

Use `Parameters<RuleMessageFunc>` instead of `Parameters<RuleMessage>` (`RuleMessage` is `string | RuleMessageFunc`) and fix the type linting to actually report such errors.

--- a/.changeset/eleven-monkeys-return.md
+++ b/.changeset/eleven-monkeys-return.md
@@ -1,0 +1,5 @@
+---
+"stylelint": patch
+---
+
+Fixed: type declarations for custom message arguments

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -15,7 +15,7 @@
 		"noUncheckedIndexedAccess": true,
 		"esModuleInterop": true,
 		"resolveJsonModule": true,
-		"skipLibCheck": true,
+		"skipLibCheck": false,
 		"typeRoots": ["./types", "./node_modules/@types"]
 	},
 	"include": ["lib", "types", "package.json"],

--- a/types/stylelint/index.d.ts
+++ b/types/stylelint/index.d.ts
@@ -353,7 +353,7 @@ declare module 'stylelint' {
 			ruleName: string;
 			result: PostcssResult;
 			message: RuleMessage;
-			messageArgs?: Parameters<RuleMessage> | undefined;
+			messageArgs?: Parameters<RuleMessageFunc> | undefined;
 			node: PostCSS.Node;
 			/**
 			 * The inclusive start index of the problem, relative to the node's

